core/shared/src/main/scala/fs2/internal/Token.scala

```diff
@@ -1,6 +1,6 @@
 package fs2.internal
 
 /** Represents a unique identifier (using object equality). */
-private[fs2] final class Token extends Serializable {
```

Comment:
I started to use it in some implementations as `Selector` to have identity for subscriber ids (i.e. topic). But perhaps is not necessary to use it, if we make PubSub Strategies private. Not sure if we want to do so, as you may write combinators with them, like for example
`priorityTopic(unboundedTopic, boundedTopic)` and that will cary on types with Token.

Reply:
That was one of my bigger comments that I haven't yet made -- specifically how much of this API should be public vs non-public, especially since it's a new API just prior to 1.0 release. We could be conservative and keep most private, giving us freedom to evolve APIs.

Reply:
I would actually use it in our production code to build thinks like `Dealer`. If PubSub will be private I cannot do so. Can we just have it in 1.0 and somehow tag it as experimental API?

Reply:
We can but then what's our philosophy for binary incompatible changes of PubSub? If folks use it and we break it, we might end up forcing folks to stay on an old version of 1.0.x.

Reply:
hm, I see. Perhaps, we can make everything protected[fs2] and then I can have wrapper for pubSubInterface that will just expose it to our code ?

Reply:
It could be even small module in fs2, which will do just that :-)

Reply:
Yeah that's a good option. I'm okay with making it public if we're confident of the API though. Give it some thought. Also, do we really need the `pubsub` package? If the strategies move to companions, I think everything can be in `fs2.concurrent` package.

Reply:
yeh we could have just PubSub and move everything to companions. 
On the visibility, How about if we make module called unstable and there we will have somethink like fs2.unstable.path_from_root_in_next_release where we will not guarantee anything ? We will explicitly say that these are ok to work and are supported, but api may change and it shouldn't ever be used by any library ?

Reply:
also the module will be published independently as part of the whole pub-sub-process

Reply:
I think we'd have a circular dependency then

Reply:
not sure. What I am thinking of is this : 
in core 
``` 
package fs2.concurrent 

protected[fs2] trait PubSub {}

```

in SBT project `unstable`

```
package fs2.unstable.concurrent 

trait PubSub extends fs2.concurrent.PubSub
```

Reply:
Hm I think that works up until the first breaking change :)

Reply:
not sure if I understand, the breakage in `core` is ok hence that is marked private, and in `unstable` we simple do not track it ?

Reply:
Okay right, the `unstable` package is just a stronger version of ScalaDoc unstable warnings then, right? You don't get any more binary compatibility and you can't necessarily use an old unstable jar with a new core or risk things like NoSuchMethodError or LinkageError.

Reply:
exactly.

Reply:
Idea is that unstable may be used in user code, but must/shall be avoided in libraries